pkg/utils/slice.go

```diff
@@ -92,3 +92,24 @@ func MuiltiGroupBy[T any, K comparable](slice []T, f func(T) []K) map[K][]T {
 	}
 	return result
 }
+
+// Returns a new slice with the element at index 'from' moved to index 'to'.
+// Does not mutate original slice.
+func MoveElement[T any](slice []T, from int, to int) []T {
```

Comment:
Down the line I want to allow moving multiple commits at once as opposed to just one, but once we implement that it'll be trivially easy to update this PR's code to handle it